xdebug_com.c

```diff
@@ -73,6 +73,9 @@ static int xdebug_create_socket_unix(const char *path TSRMLS_DC)
 		return (errno == EACCES) ? SOCK_ACCESS_ERR : SOCK_ERR;
 	}
 
+	/* Prevent the socket from being inherited by exec'd children */
+	fcntl(sockfd, F_SETFD, FD_CLOEXEC);
```

Comment:
This should check for the error condition, and print a warning (like in line 71) when the setting fails. See: https://stackoverflow.com/questions/6125068/what-does-the-fd-cloexec-fcntl-flag-do